### PR TITLE
compute AB magnitude from spectrum and bandpass

### PR DESCRIPTION
**Pull request by ntessore**
## Description

This PR contains the functionality for computing AB magnitudes from spectra and filters.

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by ntessore** [comment]
@rrjbca This is a draft for now, need to discuss the mechanism for getting spectra and bandpasses, as well as assembling, storing, and retrieving the redshift tracks if we want to use that mechanism.

File: skypy/galaxy/spectrum.py [hunk 1]
```diff
@@ -195,3 +196,82 @@ def kcorrect_spectra(redshift, stellar_mass, coefficients):
                                     wavelength.reshape(1, len(wavelength)))
 
     return wavelength_observed, sed
+
+
+def mag_ab(spec_lam, spec_flux, band_lam, band_tx, redshift=None):
```
[the remark this answers is not in the document]
Comment (ntessore) [reply skypy/galaxy/spectrum.py:201]: I do not like atypical default values for arguments, such as a scalar here for an argument that is more often than not a big array of redshift values.

[the remark this answers is not in the document]
Comment (rrjbca) [reply skypy/galaxy/spectrum.py:201]: Fair enough, although `array-like` does includes scalars, but `redshift=None` is still clear enough so I don't mind leaving this as it is

File: skypy/galaxy/tests/test_spectrum.py [hunk 2]
```diff
@@ -83,3 +84,41 @@ def test_kcorrect_spectra():
...
+    # make a wide tophat bandpass
+    th_lam = np.logspace(-10, 10, 3)
+    th_tx = np.ones(3)
+
+    # create a narrow gaussian source
+    lam = np.logspace(0, 3, 1000)
+    flam = np.exp(-((lam - 100)/10)**2)
+
+    # array of redshifts
+    z = np.arange(0., 1.01, 0.1)
```
Comment (rrjbca) [inline skypy/galaxy/tests/test_spectrum.py:118]:
```suggestion
    z = np.linspace(0, 1, 11)
```

File: skypy/galaxy/tests/test_spectrum.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import numpy as np
+import numpy.testing as npt
```
Comment (rrjbca) [inline skypy/galaxy/tests/test_spectrum.py:2]:
```suggestion
```

File: skypy/galaxy/tests/test_spectrum.py [hunk 4]
```diff
@@ -83,3 +84,41 @@ def test_kcorrect_spectra():
...
+    flam = np.exp(-((lam - 100)/10)**2)
+
+    # array of redshifts
+    z = np.arange(0., 1.01, 0.1)
+
+    # compute the AB magnitude at different redshifts
+    m = mag_ab(lam, flam, th_lam, th_tx, z)
+
+    # compare with expected redshift dependence
+    npt.assert_allclose(m, m[0] - 2.5*np.log10(1 + z))
```
Comment (rrjbca) [inline skypy/galaxy/tests/test_spectrum.py:124]:
```suggestion
    np.testing.assert_allclose(m, m[0] - 2.5*np.log10(1 + z))
```

**Pull request merged**